contrib/laldoc/laldoc/__init__.py

```diff
@@ -453,7 +363,7 @@ def append_decl(decl):
         decls = package_decl.f_public_part.f_decls
         types = {}
 
-        for decl in list(decls):
+        for decl in list(d for d in decls if d.is_a(lal.BasicDecl)):
```

Comment:
This is a violation of our coding style:
> Avoid using comprehensions as the iterated value in a regular for loop.

We can replace the above with a simple conditioned `continue` in the loop body:
```python
if not decl.is_a(lal.BasicDecl):
    continue
```